uwblueprint/starter-code-v2
# Entity CRUD GraphQL endpoint (Apollo & Express.js)

## Notion ticket link

Entity CRUD GraphQL endpoint (Apollo & Express.js) (https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d?p=2eb9fd811ad74e17b98f0984919f0801)



## Implementation description
*  GraphQL functionality for MongoDB and Postgres
*  Essentially followed the pern-graphql github repo, but made changes for typescript
*  Added a prettier line to test out 



## Steps to test
1. cd to backend directory and run command docker-compose up --build
2. Go to localhost:5000/graphql
3. Run ``` 
 mutation {
  createEntity(stringField: "A", intField: 5, stringArrayField: [], enumField: A, boolField:true) {
    id
    stringField
    intField
    stringArrayField
    enumField
    boolField
  }
}
```
4. Click the play button on the playground and verify that the it returns an id, remember that. 
5.  Run the query : 
```
query { 
	entities {
    id
    intField
    stringField
    stringArrayField
    enumField
    boolField
  }
}
```
 and verify that you got the same fields that you specified in createEntityMutation
6. Updating 
Run the mutation 
```
mutation { 
	updateEntity(id: YOUR_ID, obj:{stringField: B, intField: 21, enumField: C, 
stringArrayField:["Wow"], boolField:true}) {
    id
  }
}
```
Repeat step 5 and verify the changes have been made. 
7. Run the Mutation
```
mutation {
  deleteEntity(id: YOUR_ID)
}
```
Repeat step 5 and verify that there are no entities present.

If all steps are complete, then the graphql crud functionality works!



## What should reviewers focus on?
* entityResolvers.ts 
* entityType.ts 
* graphql.ts 
* server.ts 


## Checklist
